src/tool/cpp/cppwinrt/main.cpp

```diff
@@ -15,33 +16,70 @@ namespace xlang
 
     struct usage_exception {};
 
-    static void process_args(int const argc, char** argv)
+    static std::vector<cmd::option> options
```

Comment:
I've always wondered, why is this a static std::vector, instead of a constexpr array?

Reply:
great question

Reply:
Because for some weird reason this doesn't work:

```
std::array<cmd::option, 18> options
{
    { "input", 0 },
    { "reference", 0 },
    { "output", 0, 1 },
    { "component", 0, 1 },
````

Instead, when I switch to using an array I need to do this:

```
std::array<cmd::option, 18> options
{
    cmd::option{ "input", 0 },
    cmd::option{ "reference", 0 },
    cmd::option{ "output", 0, 1 },
    cmd::option{ "component", 0, 1 },
```

Not a high priority so haven't dug into why exactly. Uniform initialization and all that... Feel free to fix whatever's wrong.

Reply:
Instead of std::array, the terse syntax should work with plain arrays:
```c++
constexpr cmd::option[] options
{
    { "input", 0},
    { "reference", 0},
```

Reply:
Pity.